handlers/facebookapp/facebookapp.go

```diff
@@ -1568,23 +1557,34 @@ func (h *handler) BuildAttachmentRequest(ctx context.Context, b courier.Backend,
 
 var _ courier.AttachmentRequestBuilder = (*handler)(nil)
 
-type TemplateMetadata struct {
-	Templating *MsgTemplating `json:"templating"`
-}
-
 type MsgTemplating struct {
 	Template struct {
 		Name string `json:"name" validate:"required"`
 		UUID string `json:"uuid" validate:"required"`
 	} `json:"template" validate:"required,dive"`
-	Language  string   `json:"language" validate:"required"`
-	Country   string   `json:"country"`
 	Namespace string   `json:"namespace"`
 	Variables []string `json:"variables"`
 }
 
-// mapping from iso639-3_iso3166-2 to WA language code
-var languageMap = map[string]string{
+func getSupportedLanguage(lc courier.Locale) string {
+	// look for exact match
+	if lang := supportedLanguages[lc]; lang != "" {
+		return lang
+	}
+
+	// if we have a country, strip that off and look again for a match
+	l, c := lc.ToParts()
+	if c != "" {
+		if lang := supportedLanguages[courier.Locale(l)]; lang != "" {
+			return lang
+		}
+	}
+	return "en" // fallback to English
+}
```

Comment:
there 2 changes to the logic here...

 1. If we're given a code like `swa-KE`, we try that, but also try `swa`
 2. If we don't find a match, fallback on `en`... this seems to me preferable to failing

@ericnewcomer @norkans7 thoughts?

Reply:
That makes sense